kinova_driver/src/kinova_comm.cpp

```diff
@@ -135,7 +135,9 @@ KinovaComm::KinovaComm(const ros::NodeHandle& node_handle,
     {
         // If no device is specified, just use the first available device
         if (serial_number == "" || serial_number == "not_set" ||
-            std::strcmp(serial_number.c_str(), devices_list_[device_i].SerialNumber) == 0)
+            std::strncmp(serial_number.c_str(),
+                         devices_list_[device_i].SerialNumber,
+                         strlen(devices_list_[device_i].SerialNumber)-1) == 0)
```

Comment:
`devices_list_[device_i].SerialNumber` is an array of char with a fixed length. Since Serial Numbers can vary in length based on robot type and manufacturing date, comparing `strlen(devices_list_[device_i].SerialNumber)-1` characters won't work for other arms. 

I will accept this PR if you change this line to `serial_number.length()` (or maybe `min(serial_number.length(), strlen(devices_list_[device_i].SerialNumber))` to avoid overflow) instead. It should still fix your problem while maintaining compatibility with other serial numbers format

Reply:
Sorry for the late reply.
I fixed the length check according to your proposal.